gnomad_qc/analyses/compute_pext.py

```diff
@@ -0,0 +1,279 @@
+    )
+
+    return ht, tx_mt
+
+
+def get_pipeline_resources(
+    test: bool,
+    gtex_version: str,
+    vep_version: str,
+    overwrite: bool,
+) -> PipelineResourceCollection:
+    """
+    Get pipeline resources for pext annotation pipeline.
+
+    :param test: Whether to get resources for a test run.
+    :param gtex_version: GTEx version to get resources for.
+    :param vep_version: VEP version to get resources for.
+    :param overwrite: Whether to overwrite existing data.
+    :return: Pipeline resource collection for pext annotation pipeline.
+    """
+    if gtex_version == "v7":
+        gnomad_res = gnomad_grch37
+    elif gtex_version == "v10":
+        gnomad_res = gnomad_grch38
+    else:
+        raise ValueError("Invalid GTEx version")
+
+    # Initialize pipeline resource collection.
+    per_sample_pipeline = PipelineResourceCollection(
+        pipeline_name="compute_pext",
+        pipeline_resources={
+            "context HT": {"context_ht": gnomad_res.vep_context.versions[vep_version]},
+            "GTEx RSEM MT": {"tx_mt": gnomad_res.gtex_rsem.versions[gtex_version]},
+        },
+        overwrite=overwrite,
+    )
+
+    # Create resource collection for each step of the pipeline.
+    annotation_level = PipelineStepResourceCollection(
+        "--annotation-level",
+        output_resources={
+            "annotation_pext_ht": get_pext(
+                "annotation_level",
+                gtex_version,
+                vep_version,
+                test=test,
+            ),
+        },
+    )
+    base_level = PipelineStepResourceCollection(
+        "--base-level",
+        output_resources={
+            "base_pext_ht": get_pext(
+                "base_level",
+                gtex_version,
+                vep_version,
+                test=test,
+            ),
+        },
+    )
+    browser_ht = PipelineStepResourceCollection(
+        "--browser-ht",
+        pipeline_input_steps=[base_level],
+        output_resources={
+            "browser_pext_ht": get_pext(
+                "browser",
+                gtex_version,
+                vep_version,
+                test=test,
+            ),
+        },
+    )
+
+    # Add all steps to the pipeline resource collection.
+    per_sample_pipeline.add_steps(
+        {
+            "annotation_level": annotation_level,
+            "base_level": base_level,
+            "browser_ht": browser_ht,
+        }
+    )
+
+    return per_sample_pipeline
+
+
+def main(args):
+    """Script to get release pext annotation files."""
+    test = args.test
+    gtex_version = args.gtex_version
+    vep_version = args.vep_version
+    overwrite = args.overwrite
+
+    hl.init(
+        log="/pext.log",
+        default_reference="GRCh37" if gtex_version == "v7" else "GRCh38",
+        tmp_dir="gs://gnomad-tmp-4day",
+    )
+
+    pext_res = get_pipeline_resources(test, gtex_version, vep_version, overwrite)
+    ht = pext_res.context_ht.ht().naive_coalesce(5000)
```

Comment:
will this ever need to change? (will this table always have >5k partitions?)

Reply:
This is just the context HT, I was struggling to use it with the large number of partitions, and this made it much better, happy to change if you think there is a better way to handle it, or add a comment about the number of partitions in the context HTs

Reply:
I typically like to add an argument with a default (just in case the number of partitions you want to use isn't 5k in a rerun), but if that is too much for pext (which we shouldn't have to run often), just a comment sounds good to me

Reply:
I can add a default (I usually do for requested output partitions), I was just thinking that the context HT will always be about the same size, and this partitioning was really just to make it not take forever, so might not need to change the number of partitions that make it run better

Reply:
sounds good, maybe just adding a comment is the easiest then?